### PR TITLE
Jdbc analyzers refactor grouping

### PR DESCRIPTION
**Pull request by janwestphal306**
*Issue #1*

*Description of changes:*
Created FrequencyBasedAnalyzer. 
Refactored Uniqueness, Distinctness, CountDistinct.
In addition, implemented UniqueValueRatio, Entropy.

All of them make use of the FrequencyBasedAnalyzer.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sscdotopen** [comment]
Hi, its hard for me to review the PR, because it shows changes between different commits of yours, could you change so that it shows your changes w.r.t. to deequ? E.g. all changes that you made since forking.

Thanks :)

**Comment by janwestphal306** [comment]
@sscdotopen do you want me to make a pull request to the master? In my opinion, that would be way to many changes for an accurate review.

**Comment by sscdotopen** [comment]
"too many changes for a review" does not sound good :/ I'm happy to have a look at your code but you need to show me some piece that I can reasonably understand, I cannot review differences between intermediate commits of yours.

You should work on some logical piece of code, create branch for it, have it reviewed and then merge it back.

**Comment by janwestphal306** [comment]
That's unfortunate, because for now we are working on the fork and are supposed to merge to the original repo when we are finished with the jdbc-analyzers. Should we make regular pull request to original repo so you can review whole sections of code? But then we would merge our pull request in our master and have your review later.

In addition there will always be refactorings (like in this PR) so we can't prevent changes between different commits.

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcUniqueness.scala [hunk 1]
```diff
@@ -16,75 +16,24 @@
...
-    state match {
-      case Some(theState) =>
-        metricFromValue(theState.metricValue(), "Uniqueness", column, Entity.Column)
-      case _ =>
-        toFailureMetric(new EmptyStateException(
-          s"Empty state for analyzer JdbcUniqueness all input values were NULL."))
+  override def calculateMetricValue(state: JdbcFrequenciesAndNumRows): DoubleMetric = {
+    if (state.frequencies.isEmpty) {
+      return toFailureMetric(new EmptyStateException(
+        s"Empty state for analyzer JdbcDistinctness, all input values were NULL."))
```
Comment (mschroederi) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcUniqueness.scala]: `JdbcUniqueness `

**Pull request merged**